### PR TITLE
Init

### PR DESCRIPTION
**Pull request by avfletch**
This PR introduces the JSON Client, Request and Response that will form the basis of any API that both expects and returns JSON.

### CODE REVIEW

File: test/JsonClientTest.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private MockObject & ResponseInterface $response;
+
+    public function setUp(): void
+    {
+        $this->mockDelegate();
+        $this->json = self::createMock(JsonEncoder::class);
+        $this->request = self::createMock(JsonRequest::class);
+        $this->client = new JsonClient(
+            $this->delegate,
+            $this->json,
```
Comment (yevhen-minenko) [inline test/JsonClientTest.php]: I guess we can(?) agree to use named args everywhere.

**Review by yevhen-minenko — APPROVED** [review]
Looks good! The only thing that came to my mind about the general approach. When we have a lot of dependencies like this in a big project we might have a problem with versioning. Imagine a project depends on leverage-php/client and leverage-php/foo-client and they have different Guzzle versions for some reason. I guess Composer can't resolve that and we'll need to keep both up to date.

File: test/JsonClientTest.php (hunk 1 above)
Comment (avfletch) [reply test/JsonClientTest.php]: I agree for most cases. Eg I really like it for the case above.

I think there's a line to look out for.

Extreme example -

```$foo = $this->foos->findById(
    id: $id,
);```

I'm not sure whether it's a case of "single arguments don't require names" or "contextual method names don't require param names" or ... ?

**Comment by avfletch** [comment]
> I guess Composer can't resolve that and we'll need to keep both up to date.

If we follow Semantic versioning principles Composer can do a pretty good job.  But yes, I expect we'll want to keep stuff like this a lot more up-to-date than we have done in the past.  That's part of the point though.  I want to set a policy like "first of the month we update minor versions, once a quarter we take major versions" or something like that.

File: test/JsonClientTest.php (hunk 1 above)
Comment (yevhen-minenko) [reply test/JsonClientTest.php]: Oh right. Probably no need to use it for most of single arguments:
```
$foo->get($id);
$foo->findByAccount($account);
$foo->equalsTo($bar);
```

**Pull request merged**